### PR TITLE
Allow to pass env vars for local preview

### PR DESCRIPTION
**Pull request by tomasmik**

### CODE REVIEW

**Comment by Clickup-user** [comment]
Task linked: [CU-862j7j7mq Add --tf-env-var/--env-var flags to spacectl stack local-preview](https://app.clickup.com/t/862j7j7mq)

File: internal/cmd/stack/flags.go [hunk 1]
```diff
@@ -114,3 +114,13 @@ var flagShowLabels = &cli.BoolFlag{
...
 }
+
+var flagEnvVarsTF = &cli.StringSliceFlag{
+	Name:  "tf-env-vars",
+	Usage: "[Optional] Terraform environment variables injected into the run at runtime, they will be prefixed with TF_ by default, example: --tf-env-var 'foo=bar,bar=baz'",
+}
+
+var flagEnvVars = &cli.StringSliceFlag{
+	Name:  "env-vars",
+	Usage: "[Optional] Environment variables injected into the run at runtime, example: --tf-env-var 'foo=bar,bar=baz'",
```
Comment (Yantrio) [inline internal/cmd/stack/flags.go:125]: the example should use `--env-var` here
```suggestion
	Usage: "[Optional] Environment variables injected into the run at runtime, example: --env-var 'foo=bar,bar=baz'",
```

File: main.go [hunk 2]
```diff
@@ -25,10 +25,11 @@ func main() {
 		log.Fatalf("Could not parse compilation date: %v", err)
 	}
 	app := &cli.App{
-		Name:     "spacectl",
-		Version:  version,
-		Compiled: compileTime,
-		Usage:    "Programmatic access to Spacelift GraphQL API.",
+		Name:                 "spacectl",
+		Version:              version,
+		EnableBashCompletion: true,
```
Comment (Yantrio) [inline main.go:30]: Nice addition

**Review by Yantrio — CHANGES_REQUESTED** [review]

**Review by Yantrio — APPROVED** [review]

File: internal/cmd/stack/flags.go [hunk 3]
```diff
@@ -114,3 +114,13 @@ var flagShowLabels = &cli.BoolFlag{
...
+
+var flagEnvVarsTF = &cli.StringSliceFlag{
+	Name:  "tf-env-vars",
+	Usage: "[Optional] Terraform environment variables injected into the run at runtime, they will be prefixed with TF_ by default, example: --tf-env-var 'foo=bar,bar=baz'",
+}
+
+var flagEnvVars = &cli.StringSliceFlag{
+	Name:  "env-vars",
+	Usage: "[Optional] Environment variables injected into the run at runtime, example: --env-var 'foo=bar,bar=baz'",
+}
```
Comment (cube2222) [inline internal/cmd/stack/flags.go:126]: Maybe we could call them --tf-env-var-override and --env-var-override, respectively, to emphasise the overriding part?

File: internal/cmd/stack/local_preview.go [hunk 4]
```diff
@@ -125,3 +133,51 @@ func localPreview() cli.ActionFunc {
...
+// EnvironmentVariable represents a key-value pair of environment variables
+type EnvironmentVariable struct {
+	Key   graphql.String `json:"key"`
+	Value graphql.String `json:"value"`
+}
+
+func parseEnvVariablesForLocalPreview(cliCtx *cli.Context) ([]EnvironmentVariable, error) {
+	envVars := make([]EnvironmentVariable, 0)
+
+	if !cliCtx.IsSet(flagEnvVars.Name) && !cliCtx.IsSet(flagEnvVarsTF.Name) {
```
Comment (cube2222) [inline internal/cmd/stack/local_preview.go:146]: Don't think this early return is necessary, no? It'll just iterate over two empty lists.

**Review by cube2222 — APPROVED** [review]

**Pull request merged**